Automattic/jetpack
# Search: Show indexing progress in Performance Settings and Customizer

Fixes #15383.

<img width="1057" alt="Screen Shot 2020-04-17 at 3 23 10 PM" src="https://user-images.githubusercontent.com/4044428/79615176-5d42f700-80bf-11ea-8e4e-daf0ae15af0c.png">

<img width="296" alt="Screen_Shot_2020-04-21_at_4_47_48_PM" src="https://user-images.githubusercontent.com/4044428/79922572-5f73c100-83f1-11ea-8d47-c45afdb79c95.png">

#### Changes proposed in this Pull Request:
* Shows Jetpack Sync progress in Performance Settings (`/wp-admin/admin.php?page=jetpack#/performance`).
* Shows a notice in the Jetpack Search section of the customizer if sync hasn't yet finished.
* Adds a local API route returning the current Jetpack Sync status (`jetpack/v4/sync-progress`).
* Adds sync-progress state handling in Redux.
* Ports a ProgressBar component from Calypso.
* Adds Webpack configuration for a separate Jetpack Search Customizer JS bundle.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This change adds a progress bar inside the Performance Settings page to indicate indexing progress for the customer's site.
* This change adds a notice to the Jetpack Search section of the customizer to indicate indexing progress for the customer's site.

#### Testing instructions:
* Apply this patch to your Jetpack installation.
* Ensure that you have a Jetpack Search product subscription for your site.
* Navigate to the Performance Settings page for Jetpack (`/wp-admin/admin.php?page=jetpack#/performance`). Ensure that you can see a progress bar indicating search indexing status.

#### Proposed changelog entry for your changes:
* Indicate indexing progress to Jetpack Search customers in Performance Settings and in the Customizer.
